prow/cmd/pipeline/controller.go

```diff
@@ -412,7 +432,25 @@ func reconcile(c reconciler, key string) error {
 		case !ok, v != "true":
 			return nil
 		}
+
+		logrus.Infof("Delete PipelineResources for PipelineRun/%s", key)
+
+		for _, resourceBinding := range p.Spec.Resources {
+			res, err := c.getPipelineResource(ctx, namespace, resourceBinding.ResourceRef.Name)
+
+			if err != nil {
+				return fmt.Errorf("delete pipelineresource: %v", err)
+			}
+
+			if label, ok := res.Labels[kube.CreatedByProw]; ok && label == "true" {
```

Comment:
You can just do a ` res.Labels[kube.CreatedByProw] == "true"` check, this doesn't panic if the map is nil